plugins/auth-backend-module-vmware-cloud-provider/src/authenticator.ts

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  createOAuthAuthenticator,
+  decodeOAuthState,
+  encodeOAuthState,
+  OAuthState,
+  PassportOAuthAuthenticatorHelper,
+  PassportOAuthDoneCallback,
+  PassportProfile,
+} from '@backstage/plugin-auth-node';
+import { decodeJwt } from 'jose';
+import {
+  Metadata,
+  StateStoreStoreCallback,
+  StateStoreVerifyCallback,
+  Strategy as OAuth2Strategy,
+} from 'passport-oauth2';
+
+/** @public */
+export interface vmwareCloudAuthenticatorContext {
+  organizationId?: string;
+  providerStrategy: OAuth2Strategy;
+  helper: PassportOAuthAuthenticatorHelper;
+}
+
+/** @public */
+export type vmwarePassportProfile = PassportProfile & {
```

Comment:
Then it would be `VMwarePassportProfile` , no?

Reply:
No, no extra uppercase chars, we do for example `GithubBlah`, not `GitHubBlah` (except in some places that are old / accidental ...)

Reply:
I do kinda prefer the look of `VMwarePassportProfile`, but I'd just go with what @freben says; doesn't seem worth the debate since this part is not super visible anyway